src/pke/lib/scheme/bfvrns/bfvrns-cryptoparameters.cpp

```diff
@@ -490,6 +500,23 @@ void CryptoParametersBFVRNS::PrecomputeCRTTables(KeySwitchTechnique ksTech, Scal
                     m_negRlQHatInvModqPrecon[l - 1][i] = m_negRlQHatInvModq[l - 1][i].PrepModMulConst(moduliQ[i]);
                 }
             }
+
+            // Scenario when we go from Q_l to P_l
+            m_negRlQlHatInvModq.resize(sizeR);
```

Comment:
this should be changed to call reserve() and push_back(). See my comments for a similar case above.
in this case 
- the line "for (usint l = sizeR; l > 0; l--)" should become "for (size_t l = sizeR-1; l >= 0; l--)"
- negRlQlHatInvModq/m_negRlQlHatInvModqPrecon get populated from temp vectors

Reply:
I am not going to make this change as the efficiency effect is negligible (this is an offline operation). The readability is easier with the current approach.